cmd/kubeadm/app/cmd/upgrade/plan.go

```diff
@@ -60,8 +61,11 @@ func RunPlan(parentFlags *cmdUpgradeFlags) error {
 		return err
 	}
 
-	// Define Local Etcd cluster to be able to retrieve information
-	etcdCluster := kubeadmutil.LocalEtcdCluster{}
+	e := upgradeVars.cfg.Etcd
```

Comment:
What do you think about defaulting to a local etcd if no etcd configuration is supplied?

Reply:
Afaik kubeadm currently is not explicit about local etcd because e.g. the code assumes that `len(etcd.endpoints)>0` is signal for external etcd (while technically also local etcd has an endpoint).
If we move to an explicit declaration of local etcd as well, IMO we should be consistent in all the kubeadm code base

Reply:
We've had discussions before about how sharing the same Etcd config struct across different Etcd implementations has caused smell throughout the code.

There really should be a more clear configuration than `len(cfg.Etcd.Endpoints)>0` for switching between External and StaticPod etcd.